src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -1328,6 +1328,34 @@ private static bool TryValidateUtf8(ArraySegment<byte> arraySegment, bool endOfM
             return true;
         }
 
+        internal static void ThrowIfInvalidState(WebSocketState currentState, bool isDisposed, WebSocketState[] validStates)
```

Comment:
Rather than duplicating this function, can it be added to a shared class in Common?